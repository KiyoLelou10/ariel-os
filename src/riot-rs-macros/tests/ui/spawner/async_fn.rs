#![no_main]
#![feature(used_with_arg)]

// As the macro will fail, this import will not get used
#[allow(unused_imports)]
use riot_rs::asynch::Spawner;

// FAIL: spawner functions cannot be async
#[riot_rs::spawner(autostart)]
async fn main(spawner: Spawner) {}
