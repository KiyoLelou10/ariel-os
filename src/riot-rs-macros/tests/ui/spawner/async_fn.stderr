error: custom attribute panicked
 --> tests/ui/spawner/async_fn.rs:9:1
  |
9 | #[riot_rs::spawner(autostart)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: spawner functions cannot be async, consider using `task` instead
